coral-common/src/main/java/com/linkedin/coral/common/ToRelConverter.java

```diff
@@ -177,8 +177,10 @@ protected RelBuilder getRelBuilder() {
     // OR, AND, CASE clauses and simplify those. This has two problems:
     // 1. Our type system is not perfect replication of Hive so this can be incorrect
     // 2. Converted expression is harder to validate for correctness(because it appears different from input)
-    Hook.REL_BUILDER_SIMPLIFY.add(Hook.propertyJ(false));
-    relBuilder = RelBuilder.create(config);
+    if (relBuilder == null) {
+      Hook.REL_BUILDER_SIMPLIFY.add(Hook.propertyJ(false));
+      relBuilder = RelBuilder.create(config);
+    }
```

Comment:
Do you verify that there is no side effect on this change? You are refactoring `relBuilder` to an singleton here.

Reply:
Emm, all the tests including unit tests and i-tests are not affected.
I made this change because `relBuilder` is a class-wide field, if it is not singleton, it doesn't need to be a class-wide field.